test/test_sssd_conf.py

```diff
@@ -0,0 +1,189 @@
+"""
+This module contains unit tests for sssd_conf.py and related functions
+
+The unit test name reflects mane of tested sssd_conf method. For example
+'test_create_template_exists' is unit test for method 'create' of sssd_conf.
+"""
+import configparser
+import os
+import shutil
+import pytest
+import pathlib
+from pathlib import Path
+from configparser import ConfigParser
+from SCAutolib.models.sssd_conf import SSSDConf
+
+DIR_PATH = os.path.dirname(os.path.abspath(__file__))
+FILES_DIR = os.path.join(DIR_PATH, "files")
+
+
+def load_file_to_parser(config_file: pathlib.Path):
+    """
+    Instantiate parser object and update it with content of config file
+
+    :param config_file: path to config file
+    :return: parser object
+    """
+    parser = ConfigParser()
+    parser.optionxform = str
+    with config_file.open() as f1:
+        parser.read_file(f1)
+    return parser
+
+
+def parser_object_to_dict(parser_object: configparser.ConfigParser):
+    """
+    Transform content of parser object to dictionary
+
+    :param parser_object: object to be transformed
+    :return: dictionary reflecting content of parser object
+    """
+    content = {section: dict(parser_object.items(section)) for section in parser_object.sections()}
+    return content
+
+
+def compare_conf_files(file1: pathlib.Path, file2: pathlib.Path):
+    """
+    Compare content of two config files ignoring empty lines and reshuffling
+    of content
+
+    :param file1:  first of compared files
+    :param file2:  second of compared files
+    :return: True if content of files matches, False otherwise
+    """
+    dict1 = parser_object_to_dict(load_file_to_parser(file1))
+    dict2 = parser_object_to_dict(load_file_to_parser(file2))
+    assert dict1 == dict2
+
+
+# TESTS
+def test_create_template_exists():
+    """
+    Template file exists
+    """
+    assert SSSDConf._template.is_file()
+
+
+def test_check_backups(tmpdir):
+    """
+    Exception is raised if backup file exists
+    """
+    tmpfile = Path(tmpdir, "testfile")
+    with tmpfile.open('w') as original:
+        original.write('Testing backup original')
```

Comment:
No need to write something to files, `tmpfile.touch()` would be enough (just create the empty file)